heron/spi/src/java/com/twitter/heron/spi/utils/TMasterUtils.java

```diff
@@ -114,49 +114,51 @@ private static boolean sendGetRequest(URL endpoint, String command,
    */
   private static TopologyAPI.TopologyState getRuntimeTopologyState(
       String topologyName,
-      SchedulerStateManagerAdaptor statemgr) {
+      SchedulerStateManagerAdaptor statemgr) throws TMasterException {
     PhysicalPlans.PhysicalPlan plan = statemgr.getPhysicalPlan(topologyName);
 
     if (plan == null) {
-      LOG.log(Level.SEVERE, "Failed to get physical plan for topology {0}", topologyName);
-      return null;
+      String errMsg = String.format("Failed to get physical plan for topology '%s'", topologyName);
+      throw new TMasterException(errMsg);
```

Comment:
Wondering if we could use `Preconditions.checkState` throwing `IllegalStateException` instead of `TMasterException `